src/js/utils/SchemaFormUtil.js

```diff
@@ -1,15 +1,47 @@
 import tv4 from 'tv4';
 
+import FormUtil from './FormUtil';
+import Util from './Util';
+
 function filteredPaths(combinedPath) {
   return combinedPath.split('/').filter(function (path) {
     return path.length > 0;
   });
 }
 
-function setDefinitionValue(thingToSet, definition) {
+function setDefinitionValue(thingToSet, definition, renderRemove) {
   let {path, value} = thingToSet;
   let definitionToSet = SchemaFormUtil.getDefinitionFromPath(definition, path);
 
+  if (Array.isArray(value)) {
+    let prop = path[path.length - 1];
+
+    let firstIndex = 0;
+    definitionToSet.definition.find(function (field, i) {
+      if (FormUtil.isFieldInstanceOfProp(prop, field)) {
+        firstIndex = i;
+        return true;
+      }
+
+      return false;
+    });
+
+    FormUtil.removePropID(definitionToSet.definition, prop);
```

Comment:
Ok, than let's adjust the name to something "meaningful"... What about 
`FormUtil.removeEmptyFields` or something similar?